### PR TITLE
Fixed a compiler warning about unsafe variable usage

### PR DESCRIPTION
**Pull request by vfsoraki**
The warning is at `lib/mongo_ecto/normalized_query.ex:86`, and the variable `pipeline`.

I fixed it by using an anonymous function at the end of pipe operator.

### CODE REVIEW

File: lib/mongo_ecto/normalized_query.ex [hunk 1]
```diff
@@ -78,13 +78,10 @@ defmodule Mongo.Ecto.NormalizedQuery do
         {:skip,  value} -> ["$skip":  value]
       end)
       |> Kernel.++(pipeline)
+      |> (fn pipeline -> if query != %{}, do: [["$match": query] | pipeline], else: pipeline end).()
```
Comment (michalmuskala) [inline lib/mongo_ecto/normalized_query.ex]: Let's extract it into a private function, anonymous functions in the pipeline like this are not a great idea in terms of readability.

File: lib/mongo_ecto/normalized_query.ex [hunk 2]
```diff
@@ -78,13 +78,10 @@ defmodule Mongo.Ecto.NormalizedQuery do
...
 
-    if query != %{} do
-      pipeline = [["$match": query] | pipeline]
-    end
-
-    %AggregateQuery{coll: coll, pipeline: pipeline, pk: pk, fields: fields,
-                    database: original.prefix}
+      %AggregateQuery{coll: coll, pipeline: pipeline, pk: pk, fields: fields,
+                      database: original.prefix}
```
Comment (michalmuskala) [inline lib/mongo_ecto/normalized_query.ex]: Why the change in indentation?

File: lib/mongo_ecto/normalized_query.ex (hunk 1 above)
Comment (vfsoraki) [inline lib/mongo_ecto/normalized_query.ex]: Yes probably. I should have done that. I will make the commit in some hours.

File: lib/mongo_ecto/normalized_query.ex (hunk 2 above)
Comment (vfsoraki) [inline lib/mongo_ecto/normalized_query.ex]: Ah, the editor probably. I did not notice it.

**Comment by vfsoraki** [comment]
OK, here is the requested change.

**Comment by michalmuskala** [comment]
Thank you! ❤️
